### PR TITLE
Draft for 'Typed records in erlang header files'

### PR DESCRIPTION
**Pull request by jcc333**
Allegedly closes [#1103](https://github.com/gleam-lang/gleam/issues/1103).

A couple of notes:

1. Running on fedora in `test/language`, `make` yields the following eventual error for this branch as well as for `main`. Reading between the lines in some other issues, it seems like `case` may just not be there in the JS target yet, in which case, apologies for the redundant report.
```
cargo run -- compile-package --name language --src src --out target-javascript --target javascript
    Finished dev [unoptimized + debuginfo] target(s) in 0.10s
     Running `/home/jclemer/gleam/target/debug/gleam compile-package --name language --src src --out target-javascript --target javascript`

error: Unsupported feature for compilation target

```
2. There's some regretable/verbose test set-up in src/erl/tests.rs

### CODE REVIEW

File: src/docs.rs [hunk 1]
```diff
@@ -17,7 +17,7 @@ use askama::Template;
 use itertools::Itertools;
 use std::path::Path;
 
-const MAX_COLUMNS: isize = 65;
+pub const MAX_COLUMNS: isize = 65;
```
Comment (jcc333) [inline src/docs.rs:20]: This seems like it might be a poor change to make; I'd be happy to take an alternate route

Comment (lpil) [reply src/docs.rs:20]: I think the use of a constant is good, but for the other Erlang we're using 80.

https://github.com/gleam-lang/gleam/blob/6d2ce262451a268739c2e44c1f42a6df825c57d5/src/erl.rs#L401

Let's extract that into a constant and share it between these two places? Keeping the docs one private.

**Review by lpil — COMMENTED** [review]
Very nice! I'm surprised by how little code this is as well.

I think we may need to run `cargo fmt` to make CI happy.

We could improve this to have good formatting of long lines by converting the string building part of `record_definition` to use the printing algebra, though this could be a later PR.

**Comment by jcc333** [comment]
- I rewrote history to take out the use of `docs::MAX_COLUMNS`
- "converting the string building part of record_definition to use the printing algebra" I may need a hint as to what that would entail - I'm not entirely sure I follow

**Comment by lpil** [comment]
> "converting the string building part of record_definition to use the printing algebra" I may need a hint as to what that would entail - I'm not entirely sure I follow

Yes it's a bit unclear! Let's merge this in and if you're interested in adding full formatting support let me know and I'll explain more.

**Comment by lpil** [comment]
Oh, and would you mind updating the changelog file? Thanks

**Comment by lpil** [comment]
I'm gunna merge this. Thank you!!

**Pull request merged**